### PR TITLE
feat(inputs.x509_cert): Add tag for certificate type-classification

### LINKED ISSUE #7012: x509_cert plugin - add tag to identify cert as intermediate or end
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:
Add a tag to identify if the recorded cert information is an intermediate cert or end cert
### Current behavior:
No way to easily ignore intermediate certs when checking expiry of a cert chain
### Desired behavior:
Add tag that distinguishes intermediate certs from end certs
### Use case: 
Trying to do a grafana panel showing the expiry of certs for ssl web sites is made more difficult because unless you somehow filter out intermediate certs, the expiry times of the end cert and intermediate get muddled.

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #7012

This PR classifies the type of a certificate into _leaf_, _intermediate_ and _root_ certificates. The information is added as `type` tag.

### CODE REVIEW

**Review by powersj — COMMENTED** [review]
Thank you for the helpful inline comments. Do we have test coverage for these new paths already?

**Comment by srebhan** [comment]
@powersj the problem with tests using a certificate from file is that those might expire... Let me think about a way...

**Comment by powersj** [comment]
> the problem with tests using a certificate from file is that those might expire... 

Right, we do have some have some test certs in `testutil/pki`. I am cool with adding some certs there as long as you document how they are created.

File: plugins/inputs/x509_cert/x509_cert_test.go [hunk 1]
```diff
@@ -463,3 +470,197 @@ func TestServerName(t *testing.T) {
...
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPriv.PublicKey, caPriv)
+	require.NoError(t, err)
+	caPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: caBytes})
+
+	// Write CA cert
+	f, err := os.Create(filepath.Join(tmpDir, "ca.pem"))
```
Comment (powersj) [inline plugins/inputs/x509_cert/x509_cert_test.go:506]: `path/filepath` is [OS aware](https://pkg.go.dev/path/filepath#Join) do we need to use [`path`](https://pkg.go.dev/path#Join) instead only for testing?

**Review by powersj — APPROVED** [review]
One test question. As soon as tests pass, ship it

File: plugins/inputs/x509_cert/x509_cert_test.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/x509_cert/x509_cert_test.go:506]: The idea is to use a OS-aware URI. Fixed the test by fixing the expectations (hopefully).

**Pull request merged**